XML Files

EliLentz opened the pull request:
1. Added the ability to generate bunnies in advance, using XML
2. Removed the ability to generate bunnies using the "GenerateInitialBunnies" function
3. Now the life cycle and the change of years pass automatically (every 3 seconds)
4. Fixed bug with bunny breeding. (the number of new individuals depended solely on the number of males)

EliLentz committed:
XML Files

1. Added the ability to generate bunnies in advance, using XML
2. Removed the ability to generate bunnies using the "GenerateInitialBunnies" function
3. Now the life cycle and the change of years pass automatically (every 3 seconds)
4. Fixed bug with bunny breeding. (the number of new individuals depended solely on the number of males)

EliLentz committed:
One Comment

Added comment about the function "TimesHill"

EliLentz committed:
Stopper

Added a stopper that allows you to stop the loop and output data to the console
P.S. To stop data output press "Enter"

EliLentz committed:
Comment "stopFlow"

Added cooment about value "stopFlow"

igalvino commented on Hill.cs:
```diff
@@ -34,10 +35,14 @@ public Hill(List<Bunny> bunnies)
         /// </summary>
         private void TheCycleOfLife()
```
RunCycle()*

igalvino commented on Hill.cs:
```diff
@@ -34,10 +35,14 @@ public Hill(List<Bunny> bunnies)
         private void TheCycleOfLife()
         {
-            while (_hill.Count != 0)//circle in endless rage
+            while (_hill.Count != 0)
```
while(true)

igalvino commented on IntegeratorXml.cs:
```diff
@@ -0,0 +1,111 @@
+
+namespace Integerator
+{
+    class IntegeratorXml
```
*XMLReader

igalvino commented on IntegeratorXml.cs:
```diff
@@ -0,0 +1,111 @@
+            {
+                if (childNode.Name == "Sex")
+                {
+                    attrs[0] = (int)Enum.Parse(typeof(Sex), childNode.InnerText);
```
Use TryParse instead

igalvino commented on IntegeratorXml.cs:
```diff
@@ -0,0 +1,111 @@
+                XmlNode mainAttr = xnode.Attributes.GetNamedItem("Name");
+                selectedName = mainAttr.Value;
+
+                dataBunny = SearcherLong(xnode);
```
can be removed

igalvino commented on IntegeratorXml.cs:
```diff
@@ -0,0 +1,111 @@
+        /// </summary>
+        /// <param name="xnode"></param>
+        /// <returns></returns>
+        private static int[] SearcherLong(XmlNode xnode)
```
can be removed

igalvino commented on IntegeratorXml.cs:
```diff
@@ -0,0 +1,111 @@
+                selectedName = mainAttr.Value;
+
+                dataBunny = SearcherLong(xnode);
+                dataBunny = SearcherShort(xnode);
```
Bunny ParseSingleBunny(xnode)

igalvino commented on IntegeratorXml.cs:
```diff
@@ -0,0 +1,111 @@
+                initialBunnies.Add(selectedBunny);
+            }
+
+            Logic.PrintNewBunnies(initialBunnies);
```
Dont call the print from the reader.
Reader should only do read.

igalvino commented on IntegeratorXml.cs:
```diff
@@ -0,0 +1,111 @@
+        /// <returns></returns>
+        private static int[] SearcherShort(XmlNode xnode)
+        {
+            int[] attrs = new int[3];
```
int age = 0;
string name....


name = Int32.TryParse(childAttr.InnerText);


return new Bunny(name...);

igalvino commented on Logic.cs:
```diff
@@ -10,6 +12,8 @@ public class Logic
         public static int countNewBunnies = 0;//how many bunnies were born
         public static int countDeadBunnies = 0;//how many bunnies were died
 
+        private static volatile bool stopFlow = false;//Regulates flow stopping
```
try without volatile, and return with answers.

EliLentz replied:
It's working without "volatile"

igalvino commented on Logic.cs:
```diff
@@ -123,6 +90,46 @@ public static void KillOldBunnies(List<Bunny> currentBunnies)
+
+            while (stopFlow)
+            {
+                Task taskStopper = Stopper();
```
This should only happen once, not in a loop.

igalvino commented on Program_1.cs:
```diff
@@ -9,7 +10,11 @@ static void Main(string[] args)
-            bunnies = Logic.GenerateInitialBunnies(5); //giving random features (color and gender)for the bunnies
+            //bunnies = Logic.GenerateInitialBunnies(5); //giving random features (color and gender)for the bunnies
+
+            string xmlFile = "InitailBunnies.xml"; //file's URL address
```
should be const string FILE_NAME = ...

igalvino commented on Logic.cs:
```diff
@@ -123,6 +90,46 @@ public static void KillOldBunnies(List<Bunny> currentBunnies)
+        /// </summary>
+        public static void TimesHill()
+        {
+            Thread.Sleep(3000);
```
3000 should be const

igalvino commented on Logic.cs:
```diff
@@ -123,6 +90,46 @@ public static void KillOldBunnies(List<Bunny> currentBunnies)
+        /// <summary>
+        /// This function pass a year every 3 seconds
+        /// </summary>
+        public static void TimesHill()
```
*find better name

igalvino reviewed (commented):
The comments

EliLentz committed:
Revisions

1. IntegeratorXml.cs -> XMLReader.cs
2. Changed converting bunnies from XML to List
3. Returned the GenerateInitialBunnies function, which is now called if the XML file could not be found
4. TimesHill -> RunTime
5. Slightly changed the work of the Stopper method in the function RunTime
6. TheCycleOfLife -> RunCycle

EliLentz committed:
File renamed to XMLReader
